app/controllers/api/v1/resources_controller.rb

```diff
@@ -4,51 +4,91 @@ class Api::V1::ResourcesController < ApplicationController
 
   before_filter :authenticate_api_key!
   before_filter :require_allowed_model
-
-  # FIXME find a way to document api calls
+  before_filter :require_strictly_correct_params, only: :create
 
   def index
-    filter_params = params[model_name.singularize].presence
+    filter_params = params[model_name].presence
```

Comment:
No, it is not consistent but datatables controller is not public API. `model_name` here is singular and I think it is more natural and consistent with params for create action.